extension.js

```diff
@@ -1,57 +1,177 @@
+        const lastOccuranceRequire = /require\s?\((?![\s\S]*require\s?\()/.exec(textLowerCased);
+        const defineIndex = lastOccuranceDefine && lastOccuranceDefine.index || -1;
+        const requireIndex = lastOccuranceRequire && lastOccuranceRequire.index || -1;
+        const lastOccuranceRequireOrDefine = defineIndex > requireIndex ? defineIndex : requireIndex;
+
+        return textBeforeCharWithoutComments.substr(lastOccuranceRequireOrDefine > -1 ? lastOccuranceRequireOrDefine : 0);
+    }
+
+    /**
+     * Check if string contains multiple define or require statements
+     * @param {String} str to search in 
+     * @return {Bool}
+     */
+    stringHasMultipleDefineOrRequireStatements(str) {
+        return (str.match(/^\s*(define|require)\(/gm) || []).length > 1;
+    }
+
+    /**
+     * Check if needle is a module (path) in define or require statement.
+     * @param {String} needle to search for
+     * @param {String} haystack which might contain needle
+     * @return {Bool}
+     */
+    stringIsPartOfDefineOrRequireStatement(needle, haystack) {
+        const regex = new RegExp([
+            '(define|require)\\s?\\(',
+            '(',
+                '(',
+                    '\\[',
+                        '[\\s\\S]*?',
+                        needle,
+                        '[\\s\\S]*?',
+                    '\\]',
+                ')|(',
+                    '[\\s\\S]*?',
+                    needle,
+                    '.*\\)',
+                    '(\\s?{|\\.)',
+                ')',
+            ')'
+        ].join(''), 'g');
+        return regex.test(haystack);
+    }
+
+    /**
+     * Get text containing word in range. Text is stripped of comments 
+     * if text contains multiple define/require statements only module part is returned
+     * or if its is part of define/require only the statement 
+     * 
+     * @param {VSCODE Document} document 
+     * @param {VSCODE Range} range 
+     * @return {String} text
+     */
+    getTextContainingWord(document, range) {
+        const text = this.stripAllComments(document.getText());
+        const textAtCaret = document.getText(range);
+
+        let textToParse = text;
+
+        if (this.stringHasMultipleDefineOrRequireStatements(text)) {
+            const lineContainingWordAtCaret = document.lineAt(range._start._line).text;
+
+            if (this.stringIsPartOfDefineOrRequireStatement(textAtCaret, lineContainingWordAtCaret)) {
+                textToParse = lineContainingWordAtCaret;
+            } else {
+                textToParse = this.getRequireOrDefineCodeUntilCharacterIndex(document, range._start._line, range._end._character);
+            }
+        }
+        
+        if (this.stringIsPartOfDefineOrRequireStatement(textAtCaret, textToParse)) {
+            const start = text.indexOf(textToParse);
+            const end = text.indexOf('{', start) + 1;
```

Comment:
When second `stringIsPartOfDefineOrRequireStatement` inside `getTextContainingWord` returns true, this is searching for the first `{` and this is the start of the first require/define block. When the name is used in an if statement, it breaks the flow. Using this at `const constructors = this.findConstructor(document, textAtCaret, textToParse);` results in a failure to find the constructor. I'm adding test10.js to cover this case.